packages/model-viewer/src/three-components/TextureUtils.ts

```diff
@@ -173,80 +150,65 @@ export default class TextureUtils extends EventDispatcher {
     return this.skyboxCache.get(url)!;
   }
 
-  /**
-   * Loads a WebGLRenderTarget from a given URL. The render target in this
-   * case will be assumed to be used as an environment map.
-   */
-  private loadEnvironmentMapFromUrl(
-      url: string,
-      progressTracker?: ProgressTracker): Promise<WebGLRenderTarget> {
-    if (!this.environmentMapCache.has(url)) {
-      const environmentMapLoads =
-          this.loadSkyboxFromUrl(url, progressTracker).then((equirect) => {
-            const cubeUV = this.PMREMGenerator.fromEquirectangular(equirect);
-            this.addMetadata(cubeUV.texture, url);
-            return cubeUV;
-          });
-      this.PMREMGenerator.compileEquirectangularShader();
-
-      this.environmentMapCache.set(url, environmentMapLoads);
-    }
+  private GenerateEnvironmentMap(scene: Scene) {
+    const renderer = this.threeRenderer;
+    const cubeTarget = new WebGLCubeRenderTarget(256, {
+      generateMipmaps: false,
+      type: UnsignedByteType,
+      format: RGBEFormat,
+      encoding: RGBEEncoding,
+      depthBuffer: false
+    });
+    const cubeCamera = new CubeCamera(0.1, 100, cubeTarget);
+    scene.add(cubeCamera);
+
+    const outputEncoding = renderer.outputEncoding;
+    const toneMapping = renderer.toneMapping;
+    renderer.toneMapping = NoToneMapping;
+    renderer.outputEncoding = LinearEncoding;
+
+    cubeCamera.update(renderer, scene);
+    const generatedEnvironmentMap = cubeCamera.renderTarget.texture;
+    generatedEnvironmentMap.isRenderTargetTexture = false;
+    generatedEnvironmentMap.images = [1, 1, 1, 1, 1, 1];
```

Comment:
These are hacks to get  the PMREM generated at all. @mrdoob Is there a reason these checks are in place in three.js? I  think this  [example](https://threejs.org/examples/?q=dynamic#webgl_materials_cubemap_dynamic) has the same problem, but it somehow doesn't need a PMREM? Maybe because everything is mirrored?